### ISSUE TITLE
codes concatenation

### ISSUE THREAD

**Issue by Beinabih** [chavinlo/musicgen_trainer#6]
Hi, 

Thank you for your code. It has been very helpful to me in writing my own trainer.

I think the line 
`codes = torch.cat([audio, audio], dim=0)`. [(here)](https://github.com/chavinlo/musicgen_trainer/blob/5fd56f0a34f8b3ee7d5249e28a75b7e92349323f/train.py#L151)
 should be 
`codes = torch.cat([codes, codes], dim=0)` 
otherwise you wont use your encoded codebooks, right? :)

kind regards,
Jonas

**Comment by chavinlo** [comment]
um... no...

The codebook is obtained via the `preprocess_audio` that takes both the model and audio waveform as inputs. It then encodes them with encodec (here called compression_model) and then returns them.

Here is where we call that function with model and audio waveform (tensor) as arguments and we get the codebook in return https://github.com/chavinlo/musicgen_trainer/blob/5fd56f0a34f8b3ee7d5249e28a75b7e92349323f/train.py#L139

Note that the codebook here is just one batch, we then concate it to match the shape of the condition_tensors.

**Comment by chavinlo** [comment]
Feel free to correct me if I'm wrong, because the trainer does not work for anything outside overfitting at the moment.

**Comment by Beinabih** [comment]
Ah sry for the confusion, your totally right... 

I moved the `model.compression_model.encode(wav)` out of the 
`process_audio` function since i am using the [Audiocraft Dataloader](https://github.com/facebookresearch/audiocraft/blob/c9179f87d9f560c96cf6ec4de65a8fcf374c91c5/audiocraft/data/audio_dataset.py#LL237C20-L237C20).  I think i hallucinated some of my code in your trainer.py 

Do you have any Idea why we need to concat the codebooks? it seems like we need twice the batchsize than prompts as input for the `model.lm.compute_predictions()`
